### DISCUSSION TITLE
User being created when sending magic link

### DISCUSSION THREAD

**Discussion by aunumever** [supabase/supabase#7596] [General]
I am using the `generateLink()` function on the server in order to send a user that is logging in a magic link. I want to generate the link so I can send it through a third-party mail provider, not through Supabase.

However, every time I use the function, it creates the user, which I do not want. `shouldCreateUser` does not exist as an option when`generateLink()` is called for some reason, but exists if I call `signIn()` from the front end (still sending a magic link).

Why is this the case? How can I generate a link for a user without a new user being created? This allows all incorrectly entered emails and test emails to flood the db table.

**Comment by iamhectorsosa** [comment]
According to the docs, this should only be called on the server. A couple of questions:

- How are you calling this method?  
- What parameters are you using? `signup | magiclink | recovery | invite`
- `inviteUserByEmail()` seems similar, have you tried that one?

I'm interested in figuring this one out too!
